connection.go

```diff
@@ -152,20 +152,19 @@ func (ct *ConnectionTransportTLS) Dial(ctx context.Context) (
 	var conn net.Conn
 	err := runUnlessCanceled(ctx, func() error {
 		// load CA certificate
-		certs := x509.NewCertPool()
-		if !certs.AppendCertsFromPEM(ct.rootCerts) {
-			return errors.New("Unable to load root certificates")
+		config := ct.tlsConfig
+		if config == nil {
```

Comment:
this can happen also if a `nil` config is passed to `NewTLSConnectionWithTLSConfig`, which seems wrong, since in that case `ct.rootCerts` will be empty.

I suggest checking for a nil `ct.tlsConfig` here (or letting it panic in `DialWithDialer` below), and moving this cert-parsing logic into `NewTLSConnectionWithProtocols`, which will delegate to `NewTLSConnectionWithTLSConfig`.